client/src/main/java/io/pravega/client/netty/impl/FlowHandler.java

```diff
@@ -317,7 +317,7 @@ public void run() {
                     getChannel().writeAndFlush(new WireCommands.KeepAlive()).addListener(ChannelFutureListener.FIRE_EXCEPTION_ON_FAILURE);
                 }
             } catch (Exception e) {
-                log.warn("Keep alive failed, killing connection {} due to {}", connectionName, e.getMessage());
+                log.warn("Failed to send WireCommands.KeepAlive to {}. Closing this connection.", connectionName, e);
```

Comment:
Could the message be misleading at times? For example, if `getChannel()` throws `ConnectionFailedException`, technically the Keepalive wasn't sent in the first place. 

Catching specific exceptions first before catching the root `Exception` will help. It'll allow customizing the message being logged based on the specific exception being thrown.

Also, the reference to `WireCommands.KeepAlive` is avoidable. Could we just say " keep-alive"?

Reply:
Updated the log statement.
Since `ConnectionFailedException` is already logged by the Response handlers will not catch this exception specifically here. The intent of this change was to ensure we have a complete stack trace of the exception. It was observed that when the network is flaky `e.getMessage()` can be empty.(This was not helpful to infer the reason for KeepAlive failures.)